cirq/google/api/v2/program.proto

```diff
@@ -198,3 +216,81 @@ message ArgFunction {
   // The arguments to the function.
   repeated Arg args = 2;
 }
+
+// An operation that describes a circuit.
+message CircuitOperation {
+  // The index of the circuit in the top-level constant table.
+  int32 circuit_constant_index = 1;
+
+  // Specifier for repetitions of the circuit, including number of reps and
+  // repetition IDs.
+  RepetitionSpec repetition_spec = 2;
+
+  // Map of qubits in the circuit to the qubits used in this operation.
+  QubitMapping qubit_map = 3;
+
+  // Map of measurement keys in the circuit to the measurement keys used in
+  // this operation.
+  MeasurementKeyMapping measurement_key_map = 4;
+
+  // Map of args in the circuit to the values (or replacement args) used in
+  // this operation.
+  ArgMapping arg_map = 5;
+}
+
+// A description of the repetitions of a subcircuit.
+message RepetitionSpec {
+  // An ordered list of IDs for a sequence of repetitions.
+  message RepetitionIds {
```

Comment:
Do we think we will eventually change this or add onto it?

Reply:
It's possible, given that we haven't decided on an implementation for `while` loops or variable-length `for` loops.

Come to think of it, this should probably be a `oneof` with `rep_count`, since only one of them is necessary to specify repetitions (`rep_count` implies default repetition ids)